### PR TITLE
Fix - TypeError - accessing None when dequeued result is None (when timeout=None, e.g. in burst mode)

### PR DESCRIPTION
**Pull request by ronlut**
result might be None if burst=True for example

added a test (failing before the fix)

(This was introduced in #1752)

### CODE REVIEW

**Comment by lowercase00** [comment]
Thanks for catching that @ronlut !

File: tests/test_worker.py [hunk 1]
```diff
@@ -634,6 +634,15 @@ def test_timeouts(self):
...
         self.assertIn('JobTimeoutException', as_text(res.exc_info))
 
+    def test_dequeue_job_and_maintain_ttl_non_blocking(self):
+        """Not passing a timeout should return immediately with None as a result"""
+        q = Queue()
+        w = Worker([q])
+
+        # Put it on the queue with a timeout value
+        res = w.dequeue_job_and_maintain_ttl(None)
+        assert res is None
```
Comment (selwin) [inline tests/test_worker.py:644]: Mind changing this to `self.assertIsNone(w.dequeue_job_and_maintain_ttl(None))`?

**Comment by selwin** [comment]
Mind resolving the conflict so I can merge this in?

**Comment by selwin** [comment]
@ronlut can you check the tests?

**Comment by selwin** [comment]
Thanks!

**Pull request merged**